crates/ra_assists/src/merge_match_arms.rs

```diff
@@ -0,0 +1,189 @@
+use hir::db::HirDatabase;
+use ra_syntax::{
+    ast::{AstNode, MatchArm},
+    algo::non_trivia_sibling, Direction, T
+};
+use crate::{Assist, AssistCtx, AssistId};
+
+pub(crate) fn merge_match_arms(mut ctx: AssistCtx<impl HirDatabase>) -> Option<Assist> {
+    let comma = ctx.token_at_offset().find(|leaf| leaf.kind() == T![,])?;
+    let prev = non_trivia_sibling(comma.clone().into(), Direction::Prev)?;
+    let next = non_trivia_sibling(comma.clone().into(), Direction::Next)?;
+
+    if let (Some(p), Some(n)) = (prev.as_node(), next.as_node()) {
+        if let (Some(p), Some(n)) = (MatchArm::cast(p.clone()), MatchArm::cast(n.clone())) {
+            // Don't try to handle arms with guards for now - can add support for this later
+            if p.guard().is_some() || n.guard().is_some() {
+                return None;
+            }
+
+            let (p_expr, n_expr) = (p.expr()?, n.expr()?);
+
+            // Check for match arm equality by comparing lengths and then string contents
+            if p_expr.syntax().text_range().len() != n_expr.syntax().text_range().len() {
+                return None;
+            }
+
+            let p_text = p_expr.syntax().text().to_string();
+            let n_text = n_expr.syntax().text().to_string();
+
+            if p_text != n_text {
```

Comment:
Can you compare the syntax tree instead? This would fail when one of the arms for example has an extra space or `=> { ... }` instead of `=> ...`.

Reply:
I was planning on just following the suggestion from https://github.com/rust-analyzer/rust-analyzer/issues/1250#issuecomment-490057831

I'll have a look at comparing syntax trees, but this seems useful even with just the text comparison.

Reply:
Could you update rowan to 0.6.1 (https://github.com/rust-analyzer/rowan/pull/25)? Than, you'll be able to use just `(...).text() == (...).text()` for comparison, without going via `to_string()`